src/hooks/use-blog-posts.tsx

```diff
@@ -15,36 +17,35 @@ export function useBlogPosts(
   { id, slug, page, first, postCategoryId, searchTerm }: BlogPostHookParams,
   fields?: BlogPostFields[]
 ): any {
+  const [state, setState] = useState<HookData>({
+    loading: false,
+    data: null,
+    error: null
+  })
+
   function getOne() {
-    let result = {
-      data: null,
-      error: null
-    }
+    setState({ loading: true })
```

Comment:
Se for seguir por esse caminho de setar o loading como true somente no método, Talvez o correto seja manter o restante do state (até pra manter o padrão dos outros hooks).

Dessa forma:
`setState(state => ({ ...state, loading: true }))`